plugins/BEdita/Core/src/Shell/StreamsShell.php

```diff
@@ -76,4 +95,100 @@ public function removeOrphans()
         }
         $this->out(sprintf('%d stream(s) deleted', $count));
     }
+
+    /**
+     * Re-read streams metadata from file and update information in database.
+     *
+     * @return void
+     */
+    public function refreshMetadata()
+    {
+        $query = $this->Streams->find('all');
+        if ((bool)$this->param('force') === false) {
+            $query = $query->where(function (QueryExpression $exp): QueryExpression {
+                return $exp->or_(function (QueryExpression $exp): QueryExpression {
+                    return $exp
+                        ->eq($this->Streams->aliasField('file_size'), 0)
+                        ->isNull($this->Streams->aliasField('width'))
+                        ->isNull($this->Streams->aliasField('height'));
+                });
+            });
+        }
+
+        $count = $query->count();
+        $this->info(sprintf('Approximately %d streams to be processed', $count));
+        $success = 0;
+
+        foreach ($this->streamsGenerator($query) as $stream) {
+            if ($this->updateStreamMetadata($stream)) {
+                $success++;
+            }
+        }
+
+        $this->info(sprintf('Refresh completed: %d streams updated successfully, %d failed', $success, ($count - $success)));
+    }
+
+    /**
+     * Update stream metadata.
+     *
+     * @param Stream $stream The stream to update
+     * @return bool Success status of the operation
+     */
+    protected function updateStreamMetadata(Stream $stream): bool
+    {
+        try {
+            // Read current file's content...
+            $content = $stream->contents;
+            if ($content === null) {
+                $this->warn(sprintf('  stream %s (object %d) is empty or could not be read', $stream->uuid, $stream->object_id));
+
+                return false;
+            }
+
+            // ...and write it back, triggering Stream model's methods to read metadata from file
+            $stream->contents = $content;
+            if (!$this->Streams->save($stream)) {
+                $this->err(sprintf('  error updating stream %s (object %d): %s', $stream->uuid, $stream->object_id, print_r($stream->getErrors(), true)));
+
+                return false;
+            }
```

Comment:
Since we're already catching errors, I'd use `saveOrFail` to avoid this condition:

```suggestion
            $this->Streams->saveOrFail($stream));
```